src/ray/raylet/scheduling/cluster_resource_scheduler_test.cc

```diff
@@ -1100,6 +1124,14 @@ TEST_F(ClusterResourceSchedulerTest, ObjectStoreMemoryUsageTest) {
     auto available = data.resources_available();
     auto total = data.resources_total();
     ASSERT_EQ(available["object_store_memory"], 0);
+    ASSERT_EQ(resource_scheduler.GetLocalNodeResources()
```

Comment:
This can happen when we have fallback allocation and spill?

Reply:
Hmm do we need to fix this?

Reply:
Oh, this is my learning question: the test is already there and I want to make sure I understand the reason why it can happen.

Reply:
Maybe the test case cannot happen in real world.

Reply:
Ah I see, yes I think that's right. Also just good to check these boundary cases in case the object store memory reporting is buggy.

Reply:
Shouldn't we clip to 0.0?

Reply:
It seems that we do for the global reporting already:     ASSERT_EQ(available["object_store_memory"], 0);

Reply:
Yea, we do that for reporting
```
// Note: available may be negative, but only report positive to GCS.
    if (capacity.available != last_capacity.available && capacity.available > 0) {
      resources_data.set_resources_available_changed(true);
      (*resources_data.mutable_resources_available())[label] =
          capacity.available.Double();
    }
```

But seems we shouldn't do that for the actual `capacity.available` based on the comment?

Reply:
So it may be negative due to things like worker unblocking.

However, when reporting object store memory usage we control this directly and should be able to clip to zero. Could you add the zero clipping specifically for object store memory reporting?

Reply:
The above code applies to object store memory as well. So when we report object store memory, it's clipped to zero.

Reply:
Fixed.